core/partialevaluators_test.go

```diff
@@ -406,4 +409,61 @@ func TestPartialResultEvaluators(t *testing.T) {
 		require.Empty(t, query)
 		require.Empty(t, res)
 	})
+
+	t.Run("evaluates policy with custom builtins", func(t *testing.T) {
+		partialEvaluators := PartialResultsEvaluators{}
+		rondConfig := &RondConfig{
+			RequestFlow: RequestFlow{
+				PolicyName: "check_metadata",
+			},
+		}
+
+		builtinDecl := &ast.Builtin{
+			Name: "some_builtin",
+			Decl: types.NewFunction(types.Args(types.A), types.A),
+		}
+
+		opaModule := MustNewOPAModuleConfigWithBuiltins([]Module{
+			{
+				Name: "example.rego",
+				Content: `package policies
+			check_metadata {
+				some_builtin("test")
+			}`,
+			},
+		}, BuiltinDeclarations{
+			"some_builtin": builtinDecl,
+		})
+
+		someBuiltinInvoked := false
+		builtins := []func(*rego.Rego){
+			rego.Function1(
+				&rego.Function{
+					Name: "some_builtin",
+					Decl: types.NewFunction(types.Args(types.A), types.A),
```

Comment:
Maybe for the interface we could set it only once with all the necessary, and compose it accordingly? Otherwise there is a duplication between the declaration and implementation. We could also create an issue on this to improve later from my pov, it does not block this PR

Reply:
So you are suggesting re-joining the two structs and think about the proper interface later on?

Reply:
At the moment are added 2 structs in 2 different moments. My suggestion is to add only 1 struct with a fn or so on (probably a custom struct which we can arrange), to simplify the addition of custom builtin by the sdk user

Reply:
sorry I was misleaded by another comment and thought this was about the audit struct. For the builtins I agree with you, I started thinking about a way to aggregate these details but the sdk is not yet ready to let users define their own builtins. I guess when we get to that feature we will put up a single struct to define and declare builtins in a single place